internal/peering-request-operator/foreign-cluster.go

```diff
@@ -0,0 +1,129 @@
+package peering_request_operator
+
+import (
+	"errors"
+	v1 "github.com/liqoTech/liqo/api/discovery/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"k8s.io/utils/pointer"
+	"strings"
+)
+
+func (r *PeeringRequestReconciler) UpdateForeignCluster(pr *v1.PeeringRequest) error {
+	tmp, err := r.crdClient.Resource("foreignclusters").List(metav1.ListOptions{
+		LabelSelector: "cluster-id=" + pr.Spec.ClusterID,
+	})
+	if err != nil {
+		klog.Error(err, err.Error())
+		return err
+	}
+	fcList, ok := tmp.(*v1.ForeignClusterList)
+	if !ok {
+		err = errors.New("retrieved object is not a ForeignClusterList")
+		klog.Error(err, err.Error())
+		return err
+	}
+
+	if len(fcList.Items) == 0 {
+		// create it
+		fc, err := r.createForeignCluster(pr)
+		if err != nil {
+			return err
+		}
+		r.setOwner(pr, fc)
+		return nil
+	} else {
+		// update it
+		fc := &fcList.Items[0]
+		if fc.Status.Incoming.PeeringRequest != nil {
+			// already up to date
+			return nil
+		}
+		fc.Status.Incoming.PeeringRequest = &corev1.ObjectReference{
+			Kind:       pr.Kind,
+			Name:       pr.Name,
+			UID:        pr.UID,
+			APIVersion: pr.APIVersion,
+		}
+		_, err = r.crdClient.Resource("foreignclusters").Update(fc.Name, fc, metav1.UpdateOptions{})
+		if err != nil {
+			klog.Error(err, err.Error())
+			return err
+		}
+		r.setOwner(pr, fc)
+		return nil
+	}
+}
+
+func (r *PeeringRequestReconciler) createForeignCluster(pr *v1.PeeringRequest) (*v1.ForeignCluster, error) {
+	var cnf *rest.Config
+	var err error
+
+	cnf, err = pr.GetConfig(r.crdClient.Client())
+	if err != nil {
+		if strings.Contains(err.Error(), "yaml") && r.Testing {
+			klog.Warning("using test settings")
+			cnf = r.ForeignConfig
+		} else {
+			klog.Error(err, err.Error())
+			return nil, err
+		}
+	}
+
+	fc := &v1.ForeignCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pr.Spec.ClusterID,
+		},
+		Spec: v1.ForeignClusterSpec{
+			ClusterID:        pr.Spec.ClusterID,
+			Namespace:        pr.Spec.Namespace,
+			Join:             false,
+			ApiUrl:           cnf.Host,
+			DiscoveryType:    v1.IncomingPeeringDiscovery,
+			AllowUntrustedCA: pr.Spec.OriginClusterSets.AllowUntrustedCA,
+		},
+		Status: v1.ForeignClusterStatus{
```

Comment:
if I remember well, when you create an object the status is dropped so this should be ignored

Reply:
No, it is working. I tested it in my k3s installation

Reply:
Good to know!